### ISSUE TITLE
Firebase phone auth in IOS Simulator not working every alternate time

### ISSUE THREAD

**Issue by cloudcerebro** [FlutterFlow/flutterflow-issues#325]
Please **thumbs up** this issue if you have also experienced it. You may also add more information if there is something relevant that was not mentioned. However, please refrain from comments that are not constructive, like "I have this problem too", etc.

## Expected behavior (required)
Logs in to home screen after phone sign-in

## Current behavior (required)
On entering the app, the user enters the phone number and clicks Login button. A webview is opened to verify if we are a robot and on returning back to the app, it does not enter Verify code page first time.

If we are entering the phone number and click login button again, the webview opens again and on returning, it enters the verify code page.

If we sign-out and login in again, then not working. It works every alternate time.

Exception received in debugger:

flutter: Another exception was thrown: Exception: 'package:go_router/src/go_router_delegate.dart': Failed assertion: line 443 pos 12: 'matchStacks.isNotEmpty': no routes for location: /link?deep_link_id=https%3A%2F%2Fproject-mux-copy.firebaseapp.com%2F__%2Fauth%2Fcallback%3FauthType%3DverifyApp%26recaptchaToken%3D03AD1IbLD5lbIoELlZKNcGhNMuBwbwElNBCZ7vUnVM6A5V4SOtbHvD1ZxLAutMNyEJsYO20lpixw.......

It works on Android emulator.

## To Reproduce (required)
1. Download code
2. Open in VS Code
3. Install dependencies
4. Choose IOS Simulator as target
5. Run Debug
6. Enter Phone number '+919876598765' (Firebase Phone number for testing) and click Login
7. Enters Verification Webview and returns back to Login page, and stops, hence not working.
8. Again enter phone number and click Login.
9. Enters Verification Webview and returns back to Login page, then continues to VerifyCode page (Code: 111111), hence working.

### Bug Report Code (required, generate from Widget Tree)

IT4wj8mBxItOsbRe1aqMbflFmiQXCDo1U40ZkcFDQis9f5CvOpYUec/6P05pTfG4Twk7e2emp14DpsGOuIXhK+1fJSeCc4h7/5AJZTzMfGCjRbHfU/HUACkbeacOQSj6

## Context (required)
User can't login to the application the first time.

## Screenshots / recordings
[Screen recording](https://www.veed.io/view/139221e2-7d1b-4311-a445-d1c47513011c?sharingWidget=true&panel=share)

Project Link:
[Project link](https://app.flutterflow.io/project/ios-login-demo-3ewl67)

## Your environment

* Version of FlutterFlow used: 3.0 (Released on December 30, 2022 - Flutter  3.3.4
* Platform (e.g. Web, MacOS Desktop): IOS
* Browser name and version: IOS Simulator (iPhone 14 Pro Max)
* Operating system and version (desktop or mobile): iOS 16.2

**Comment by Pepino-GR** [comment]
Hi cloudcerebro,

I cannot reproduce this issue with the descripted steps to reproduce. Is there any error shown inside the Snack Bar(message bar)? Are there any API restrictions set in the API & Services Google Cloud Console?

**Comment by cloudcerebro** [comment]
Are you using the same project I linked? If so, if it works for you, and not me, how come there might be API restrictions? If you have a sample project that uses firebase phone auth, that runs in IOS simulator, can you share the project?

**Comment by Pepino-GR** [comment]
Hi cloudcerebro,

if I use your linked project, I have to clone your project and set my own firebase project to use within your project. So it doesn't help us for your project. I have verified your settings inside your project and it seems to be o.k. ( without any firebase configuration).If you do not set explicit restrictions inside your API & Services Google Cloud Console then there are no restrictions.   

Is there any error shown inside the Snack Bar(message bar)? 

As I'm unable to reproduce your issue with other projects it seems to be a support issue. Please use chat support or contact support [at] flutterflow.io for further assistance.

**Comment by cloudcerebro** [comment]
Thanks. There is no snack bar. I've mentioned the link of video showing the error message. I've also mentioned the error message in the issue. 

There is no explicit restrictions being set in API & Services in Google Cloud Console. 

I'm using a M1 mac to run the simulator. I hope you are using the same.

Is it not possible for you to just download the code and use my own firebase project? This is just a demo flutterflow project and a demo firebase project. My main project has the same error, hence tried replicating the error in a new demo project (linked) and the error remains.

If this is not an issue to be resolved, I would contact support for further assistance.

**Comment by Pepino-GR** [comment]
Does the issue occurred also on a physical iPhone?

**Comment by Pepino-GR** [comment]
Can you share your Firebase project ID?

**Comment by Pepino-GR** [comment]
I have done some troubleshooting with your project and now I can reproduce your issue. There seems to be a problem with the reCAPTCHA. Sorry for the delay, but we need to reproduce your issue, before we can start to fix it.

**Comment by cloudcerebro** [comment]
Thanks. I don't have developer account to check in a physical iPhone. My firebase ID is "project-mux-copy".

**Comment by wenkaifan0720** [comment]
Hi @agreaves. I can reproduce the bug but don't understand why it happens. I checked the firebase phone auth doc (https://firebase.google.com/docs/auth/flutter/phone-auth) and they don't mention that native platform requires reCAPTCHA in the first place. And I can't receive SMS message or test on a physical iPhone right now...

**Comment by Pepino-GR** [comment]
Hi wenkaifan,

The documentation says:
Phone number sign-in is only available for use on real devices and the web. To test your authentication flow on device emulators, please see [Testing](https://firebase.google.com/docs/auth/flutter/phone-auth?hl=en#testing)

If you use an emulator with a testnumber configured in Firebase, it will use the reCAPTCHA.

**Comment by wenkaifan0720** [comment]
Hi @Pepino-GR. I see. Thanks for the clarification. This kind of behavior is really hard to debug and should not affect the app in production (i.e. on a real iPhone). @cloudcerebro, could you test if the phone auth is working on a real iPhone? 

In the meantime, I will ask around and see if anyone on the FF team has a clue.

**Comment by Pepino-GR** [comment]
Hi @wenkaifan0720,

It works on a real iPhone without issue

**Comment by wenkaifan0720** [comment]
Hi @Pepino-GR. In that case, I'll flag this issue as low priority.

**Comment by wenkaifan0720** [comment]
Just tested that the issue is still there. It is a bit annoying for testing, but not blocking. I'll close this issue for now.
